src/types/gecko-profile.js

```diff
@@ -21,12 +21,31 @@ import type { Milliseconds, Nanoseconds } from './units';
 export type IndexIntoGeckoFrameTable = number;
 export type IndexIntoGeckoStackTable = number;
 
+// Taken from the C++
+//
+// enum class MarkerPhase : int {
+//   Instant = 0,
+//   Interval = 1,
+//   IntervalStart = 2,
+//   IntervalEnd = 3,
+// };
+export type MarkerPhase = 0 | 1 | 2 | 3;
```

Comment:
Maybe this was already discussed when working on the cpp side.

I wonder if we don't combine 2 different things in one value here:
1. what type of marker this is: instant marker, interval marker, phased marker?
2. if this is a phased marker, which phase is this?

So maybe this should be split in 2 different values.

This would very easily allow other cases, like tri-phased markers: suggested to me by @padenot, there could be:
* marker 1: when a task is queued
* marker 2: when the task starts running
* marker 3: when the task ends

This could work for network markers too (where several "phased markers" would make together for 1 request):
* one phased marker for connection (eg: start dns, start connection, start TLS)
* one for the request (starts request, ends request, starts response, ends response -- maybe this should be 2 individual markers though for that one :) )

A phased marker would work for a sequence of related events.

Reply:
So we chatted briefly about this in our standup, but I wanted to put it in writing.

I think I'd like to stick with the Instant, Interval, IntervalStart, and IntervalEnd marker situation. Then, as a follow-up we can introduce the idea of a "marker set id", which can then be used to correlate information into a logical group. The network markers can then freely emit information, and it will be collected under one single group to reconstruct the entire event chain.